File: Nodal_Discontinuous_Galerkin_Methods_Hesthaven/Codes1.1/Codes2D/Grad2D.m
function [ux,uy] = Grad2D(u);

% function [ux,uy] = Grad2D(u);
% Purpose: Compute 2D gradient field of scalar u

Globals2D;

ur = Dr*u; us = Ds*u;
ux = rx.*ur + sx.*us; uy = ry.*ur + sy.*us;
return

File: Nodal_Discontinuous_Galerkin_Methods_Hesthaven/nudg++/trunk/Src/Codes2D/Grad2D.cpp
// Grad2D.m
// function [ux,uy] = Grad2D(u);
// 2007/06/06
//---------------------------------------------------------
#include "NDGLib_headers.h"
#include "NDG2D.h"


//---------------------------------------------------------
void NDG2D::Grad2D
(
  const DMat& u,    // [in]
        DMat& ux,   // [out]
        DMat& uy    // [out]
)
//---------------------------------------------------------
{
  // function [ux,uy] = Grad2D(u);
  // Purpose: Compute 2D gradient field of scalar u

  DVec ur = Dr*u, us = Ds*u;
  ux = rx.dm(ur) + sx.dm(us); uy = ry.dm(ur) + sy.dm(us);
}
